src/main/java/frc/robot/Robot.java

```diff
@@ -611,9 +615,43 @@ private void executeShootingStateMachine() {
   }
 
   private void executeClimbingStateMachine() {
-    switch (mClimingState) {
+    switch (mClimbingState) {
       case IDLE:
+        mRobotLogger.warn("Climbing state is idle");
         break;
+      case EXTENDING:
+        //TODO: Decide if climb and retract should be the same button
+        /** Checks if the climb button has been hit again, signalling it to retract */
+        if (mOperatorInterface.getClimb()) {
```

Comment:
The problem with checks like these is getClimb() will return true whenever the climb button is down, not when it was just barely pressed. This will cause it to switch between climbing and not climbing every RobotLoop() whenever the climb button is pressed. Look at startIntake [here](https://github.com/Team138Entropy/Entropy2020/blob/1d99a6945984b584e8a23f0d8bbea6e20817ab6f/src/main/java/frc/robot/OI/OperatorInterface.java#L116) for an example of how to handle this.

Reply:
Fixed